bot/src/rpc_client_utils.rs

```diff
@@ -127,22 +126,24 @@ pub fn send_and_confirm_transactions_with_spinner(
                 last_resend = Instant::now();
             }
 
-            // Wait for the next block before checking for transaction statuses
-            set_message(
-                confirmed_transactions,
-                Some(block_height),
-                last_valid_block_height,
-                &format!("Waiting for next block, {} pending...", num_transactions),
-            );
-
-            let mut new_block_height = block_height;
-            while block_height == new_block_height {
-                sleep(Duration::from_millis(500));
-                new_block_height = rpc_client.get_block_height()?;
+            if !rpc_client.commitment().is_processed() {
```

Comment:
Can we switch to `get_block_height_with_commitment` below instead of skipping this logic?

Reply:
I'm not sure I understand. What commitment would we pass to `get_block_height_with_commitment`? `get_block_height` defaults to `rpc_client.commitment`.

The concept of this change is that if we're using a client with `processed`, we don't need to wait for another block before checking transaction statuses.

Note that this was mainly done to speed up testing, so as an alternative, we can do
```
#[cfg(not(test))]
```
instead of the check.  What do you think?

Reply:
Ah yeah if this is just for tests, let's make the logic communicate that :+1: